src/opencl_common.c

```diff
@@ -1819,14 +1821,13 @@ void opencl_find_best_lws(size_t group_size_limit, int sequential_id,
 	                                       CL_PROFILING_COMMAND_END,
 	                                       sizeof(cl_ulong), &endTime, NULL),
 	               "clGetEventProfilingInfo end");
-	numloops = (int)(size_t)(200000000ULL / (endTime - startTime));
+	cl_ulong roundup = endTime - startTime - 1;
+	numloops = (int)(size_t)((200000000ULL + roundup) / (endTime - startTime));
 
 	clear_profiling_events();
 
 	if (numloops < 1)
 		numloops = 1;
-	else if (numloops > 5)
-		numloops = 5;
 
```

Comment:
I dropped this, I'm not sure what that limitation buys us (we're talking 200ms here).

Reply:
It is acceptable. But, for very fast formats, from now on, you can run unnecessary tests (when 5 is enough).

Reply:
For very fast formats, we just get more samples (and incidentally we increase the chance of winding up the GPU to full speed).